### PR TITLE
Add admin function to merge two organizations

### PR DESCRIPTION
**Pull request by Chessray**
Admins can merge two organizations by selecting fields from both. The usability of all merge functionality has also been improved, by requiring the admin to pick all fields from either side of the merge, with clear feedback on which fields still need to be set.

Closes [AB#1010](https://dev.azure.com/ncia-anet/2aa083a5-af3d-44e1-8c7b-6e9e6b124d91/_workitems/edit/1010)

#### User changes
- None

#### Superuser changes
- None

#### Admin changes
- New menu option "Merge Organizations" with functionality
- Improved usability of all merge functionality
- Allow admin to choose which customSensitiveInformation fields get merged

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [x] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [x] Added and/or updated unit tests
  - [x] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [x] Opened debt issues for anything not resolved here

### CODE REVIEW

File: src/main/java/mil/dds/anet/database/OrganizationDao.java [hunk 1]
```diff
@@ -45,6 +45,26 @@ public Organization getByUuid(String uuid) {
     return getByIds(Arrays.asList(uuid)).get(0);
   }
 
+  @InTransaction
+  public int mergeOrganizations(final Organization loserOrganization,
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/database/OrganizationDao.java:49]: Please move this method to the end of the class.

File: src/main/java/mil/dds/anet/database/OrganizationDao.java [hunk 2]
```diff
@@ -45,6 +45,26 @@ public Organization getByUuid(String uuid) {
...
+    updateForMerge("approvalSteps", "relatedObjectUuid", winnerOrganizationUuid,
+        loserOrganizationUuid);
+    updateForMerge("taskTaskedOrganizations", "organizationUuid", winnerOrganizationUuid,
+        loserOrganizationUuid);
+    updateForMerge("positions", "organizationUuid", winnerOrganizationUuid, loserOrganizationUuid);
+    updateForMerge("authorizationGroupRelatedObjects", "relatedObjectUuid", winnerOrganizationUuid,
+        loserOrganizationUuid);
+    updateForMerge("organizationAdministrativePositions", "organizationUuid",
+        winnerOrganizationUuid, loserOrganizationUuid);
+    return deleteForMerge("organizations", "uuid", loserOrganizationUuid);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/database/OrganizationDao.java:65]:
```java
    // FIXME: update organizations.parentOrgUuid
    // FIXME: update reports.advisorOrganizationUuid
    // FIXME: update reports.interlocutorOrganizationUuid
    // FIXME: update noteRelatedObjects.relatedObjectUuid
    // FIXME: update attachmentRelatedObjects.relatedObjectUuid
    // FIXME: update emailAddresses
    // FIXME: update customSensitiveInformation.relatedObjectUuid
```

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 3]
```diff
@@ -58,7 +63,7 @@ public static void assertPermission(final Person user, final String organization
   public Organization getByUuid(@GraphQLArgument(name = "uuid") String uuid) {
     Organization org = dao.getByUuid(uuid);
     if (org == null) {
-      throw new WebApplicationException("Organization not found", Status.NOT_FOUND);
+      throw new WebApplicationException("Organization not found", NOT_FOUND);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:66]: Unrelated change, please revert.

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 4]
```diff
@@ -75,7 +80,7 @@ public Organization createOrganization(@GraphQLRootContext Map<String, Object> c
     org.setProfile(
         Utils.isEmptyHtml(org.getProfile()) ? null : Utils.sanitizeHtml(org.getProfile()));
 
-    final Person user = DaoUtils.getUserFromContext(context);
+    final Person user = getUserFromContext(context);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:83]: Unrelated change, please revert.

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 5]
```diff
@@ -127,7 +132,7 @@ public Integer updateOrganization(@GraphQLRootContext Map<String, Object> contex
...
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:135]: Unrelated change, please revert.

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 6]
```diff
@@ -174,7 +179,7 @@ private int update(Person user, Organization org, Organization existing) {
     }
 
     if (numRows == 0) {
-      throw new WebApplicationException("Couldn't process organization update", Status.NOT_FOUND);
+      throw new WebApplicationException("Couldn't process organization update", NOT_FOUND);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:182]: Unrelated change, please revert.

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 7]
```diff
@@ -208,10 +213,37 @@ private int update(Person user, Organization org, Organization existing) {
...
+    if (loserOrganization.getUuid().equals(winnerOrganization.getUuid())) {
+      throw new WebApplicationException("Cannot merge identical organizations.", BAD_REQUEST);
+    }
+  }
+
   @GraphQLQuery(name = "organizationList")
   public AnetBeanList<Organization> search(@GraphQLRootContext Map<String, Object> context,
       @GraphQLArgument(name = "query") OrganizationSearchQuery query) {
-    query.setUser(DaoUtils.getUserFromContext(context));
+    query.setUser(getUserFromContext(context));
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:246]: Unrelated change, please revert.

File: src/main/java/mil/dds/anet/resources/OrganizationResource.java [hunk 8]
```diff
@@ -208,10 +213,37 @@ private int update(Person user, Organization org, Organization existing) {
...
+          NOT_FOUND);
+    }
+    AnetAuditLogger.log("Organization {} merged into {} by {}", loserOrganization,
+        winnerOrganization, user);
+    return numberOfAffectedRows;
+  }
+
+  private void checkWhetherOrganizationsAreMergeable(final Organization winnerOrganization,
+      final Organization loserOrganization) {
+    if (loserOrganization.getUuid().equals(winnerOrganization.getUuid())) {
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/resources/OrganizationResource.java:238]: Avoid NullPointerExceptions by doing:
```java
    if (Objects.equals(DaoUtils.getUuid(loserOrganization), DaoUtils.getUuid(winnerOrganization))) {
```

File: client/src/components/MergeField.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ import styled from "@emotion/styled"
 import { ALIGN_OPTIONS, setHeightOfAField } from "mergeUtils"
 import PropTypes from "prop-types"
 import React, { useEffect, useRef, useState } from "react"
+import Checkbox from "./Checkbox"
```
Comment (gjvoosten) [inline client/src/components/MergeField.js:5]: Please import as:
```javascript
import Checkbox from "components/Checkbox"
```

File: client/src/components/MergeField.js [hunk 10]
```diff
@@ -47,6 +50,13 @@ const MergeField = ({
         <ValueBox className={className} align={align}>
           {value}
         </ValueBox>
+        {displayIntentionallyLeftBlank && (
+          <Checkbox
```
Comment (gjvoosten) [inline client/src/components/MergeField.js:54]: This should have an `onChange` handler.

File: client/src/components/PeopleTable.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import _get from "lodash/get"
+import PropTypes from "prop-types"
+import React from "react"
+import { Table } from "react-bootstrap"
+
+const PeopleTable = ({ label, people }) => {
```
Comment (gjvoosten) [inline client/src/components/PeopleTable.js:6]: This new file can be removed, it isn't used anywhere.

**Review by gjvoosten — DISMISSED** [review]

**Review by midmarch — APPROVED** [review]
Merging 2 locations:

image

Merging 2 organizations:

image

**Review by pacodelcastillolopez — APPROVED** [review]
